Plugin reports successful build despite build-phase failures

### Captchas

- [x] I have searched existing issues.
- [x] I have checked [Wiki](https://github.com/wojciech-kulik/xcodebuild.nvim/wiki).
- [x] I am not asking about LSP issues. Please use GitHub Discussions for that.
- [x] The issue also occurs when using [ios-dev-starter](https://github.com/wojciech-kulik/ios-dev-starter-nvim)

### Steps to reproduce

I've noticed the plugin misses some build errors, mainly related to code signing and build phases. It can report a build as successful even when it failed. This is especially problematic when running unit tests: after a failed build the plugin runs `xcodebuild test-without-building`, so the tests run against the last successful compilation (outdated code) and produce incorrect results.

You can reproduce the issue as follows:

1. Create a new macOS project (to avoid #285)  
2. Add a new "Run Script" build phase and enter `exit 1`  
3. Build the project with `xcodebuild`  
4. Build the project with `xcodebuild.nvim`

### Actual result

`xcodebuild` reports the following error:

```
Command PhaseScriptExecution failed with a nonzero exit code

warning: Run script build phase 'Run Script' will be run during every build because it does not specify any outputs. To address this issue, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'MacApp' from project 'MacApp')
** BUILD FAILED **
```

However, `xcodebuild.nvim` reports the build as successful.

### Expected result

`xcodebuild.nvim` should display the build error and open the build log.

### Health check


==============================================================================
xcodebuild:                                                         6 ⚠️  1 ❌

Checking xcodebuild.nvim ~
- ✅ OK xcodebuild.nvim: commit #a8b8441

Checking OS ~
- ✅ OK macOS 14+: release 24.6.0

Checking plugins ~
- ✅ OK telescope.nvim installed.
- ⚠️ WARNING fzf-lua not installed. (One picker plugin is required to interact with the plugin)
- ⚠️ WARNING snacks.nvim not installed. (Optional to use as a picker or show previews for SwiftUI, UIKit, and AppKit views)
- ✅ OK nui.nvim installed.
- ✅ OK nvim-tree installed.
- ✅ OK neo-tree installed.
- ✅ OK oil.nvim installed.
- ✅ OK nvim-dap installed.
- ✅ OK nvim-dap-ui installed.
- ✅ OK nvim-treesitter installed.
- ⚠️ WARNING nvim-treesitter: Swift parser not installed. It is required to present results for tests written using Quick framework.

Checking required dependencies ~
- ✅ OK xcodebuild: version 26.0
- ✅ OK xcodebuild: installed
- ✅ OK xcrun: installed
- ✅ OK xcode-build-server: installed

Checking optional dependencies ~
- ✅ OK pymobiledevice3: installed
- ✅ OK xcodeproj: installed
- ✅ OK xcbeautify: installed
- ✅ OK rg: installed
- ✅ OK fd: installed
- ✅ OK stdbuf: installed
- ✅ OK ruby: version 3.4.5

Checking debugger ~
- ✅ OK nvim-dap: codelldb adapter configured.
- ✅ OK nvim-dap: swift configuration found.
- ✅ OK codelldb: installed

Checking buildServer.json ~
- ✅ OK buildServer.json: found

Checking .nvim/xcodebuild/settings.json ~
- ✅ OK .nvim/xcodebuild: found
- ✅ OK .nvim/xcodebuild/settings.json: found

Checking remote debugger integration (pymobiledevice3) ~
- ❌ ERROR `pymobiledevice3` detected, but the `remote_debugger` script is not installed. (see: `:help |xcodebuild.remote-debugger`)|
- ⚠️ WARNING passwordless sudo permission for `remote_debugger` is not configured.
- ⚠️ WARNING debugging on physical devices with iOS 17+ will not work.
- ⚠️ WARNING see `:help |xcodebuild.remote-debugger`| for more information.

Checking xcodebuild-offline integration ~
- ✅ OK `xcodebuild_offline` script installed
- ✅ OK sudo: configured

--- a/lua/xcodebuild/xcode_logs/parser.lua
+++ b/lua/xcodebuild/xcode_logs/parser.lua
@@ -95,6 +95,7 @@ local lineData = {}
 local lastTest = nil
 local lastErrorTest = {}
 local testSuite = nil
+local buildPhaseName = nil
 
 -- report fields
 local testsCount = 0
@@ -357,6 +358,31 @@ local function parse_build_error(line)
 end
 
 ---@param line string
+---@see ParsedBuildGenericError
+local function parse_build_phase_error(line)
+  lineType = BUILD_ERROR
+  lineData = {
+    message = {
+      "Build phase '" .. (buildPhaseName or "Unknown") .. "' failed:",
+      line,
+    },
+  }
+
+  debug_print("detected_build_phase_error", lineData)
+end
+
+---@param line string
+---@see ParsedBuildGenericError
+local function parse_general_build_failure(line)
+  lineType = BUILD_ERROR
+  lineData = {
+    message = { line },
+  }
+
+  debug_print("detected_general_build_failure", lineData)
+end
+
+---@param line string
 ---@see ParsedTest
 ---@see ParsedTestError
 local function parse_test_error(line)
@@ -608,11 +634,22 @@ local function process_line(line)
     elseif testsCount == 0 and lineType == BEGIN then
       parse_build_error(line)
     end
+  elseif string.find(line, "PhaseScriptExecution failed") then
+    flush()
+    parse_build_phase_error(line)
+    flush()
+    buildPhaseName = nil
   elseif string.find(line, "warning:") then
     flush()
     parse_warning(line)
   elseif string.find(line, "%s*~*%^~*%s*") then
     flush(line)
+  elseif string.find(line, "^PhaseScriptExecution") then
+    buildPhaseName = string.match(line, "^PhaseScriptExecution ([^/]+) /")
+    buildPhaseName = buildPhaseName and buildPhaseName:gsub("\\", "")
+  elseif string.find(line, "** BUILD FAILED **") and util.is_empty(buildErrors) then
+    parse_general_build_failure(line)
+    flush()
   elseif string.find(line, "^%s*$") then
     -- test errors can contain multiple lines with empty lines
     -- we'll flush when we encounter finished test line.

--- a/specs/parser_spec.lua
+++ b/specs/parser_spec.lua
@@ -235,6 +235,20 @@ describe("ENSURE parse_logs", function()
         assert.are.same(expectedResult, result)
       end)
     end)
+
+    describe("because of build phase failure", function()
+      it("THEN should set build errors", function()
+        local expectedResult, result = runTestCase(24)
+        assert.are.same(expectedResult, result)
+      end)
+    end)
+
+    describe("because of BUILD FAILED detection", function()
+      it("THEN should set build error", function()
+        local expectedResult, result = runTestCase(25)
+        assert.are.same(expectedResult, result)
+      end)
+    end)
   end)
 
   --

--- a/specs/parser_test_data/tc24.log
+++ b/specs/parser_test_data/tc24.log
@@ -1,0 +1,51 @@
+Command line invocation:
+    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild build -project /Users/john/repositories/SwiftUI-Redux-Demo/ReduxDemo.xcodeproj -scheme ReduxDemo -destination id=B1982473-232A-4810-A4B0-E71E4516F39B -resultBundlePath /Users/john/repositories/SwiftUI-Redux-Demo/.nvim/xcodebuild/build.xcresult -parallelizeTargets
+
+Writing result bundle at path:
+	/Users/john/repositories/SwiftUI-Redux-Demo/.nvim/xcodebuild/build.xcresult
+
+--- xcodebuild: WARNING: Using the first of multiple matching destinations:
+{ platform:iOS Simulator, arch:arm64, id:B1982473-232A-4810-A4B0-E71E4516F39B, OS:18.5, name:iPhone 16 }
+{ platform:iOS Simulator, arch:x86_64, id:B1982473-232A-4810-A4B0-E71E4516F39B, OS:18.5, name:iPhone 16 }
+ComputePackagePrebuildTargetDependencyGraph
+
+Prepare packages
+
+CreateBuildRequest
+
+SendProjectDescription
+
+CreateBuildOperation
+
+ComputeTargetDependencyGraph
+note: Building targets in dependency order
+note: Target dependency graph (1 target)
+    Target 'ReduxDemo' in project 'ReduxDemo' (no dependencies)
+
+GatherProvisioningInputs
+
+CreateBuildDescription
+
+ClangStatCache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang-stat-cache /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator18.5.sdk /Users/john/Library/Developer/Xcode/DerivedData/SDKStatCaches.noindex/iphonesimulator18.5-22F76-d5fc8ad4295d2ef488fb7d0f804ce0c4.sdkstatcache
+    cd /Users/john/repositories/SwiftUI-Redux-Demo/ReduxDemo.xcodeproj
+    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang-stat-cache /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator18.5.sdk -o /Users/john/Library/Developer/Xcode/DerivedData/SDKStatCaches.noindex/iphonesimulator18.5-22F76-d5fc8ad4295d2ef488fb7d0f804ce0c4.sdkstatcache
+
+warning: Run script build phase 'MY CUSTOM PHASE' will be run during every build because it does not specify any outputs. To address this issue, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'ReduxDemo' from project 'ReduxDemo')
+PhaseScriptExecution MY\ CUSTOM\ PHASE /Users/john/Library/Developer/Xcode/DerivedData/ReduxDemo-gttjropgetipjmadcueprkkeczpt/Build/Intermediates.noindex/ReduxDemo.build/Debug-iphonesimulator/ReduxDemo.build/Script-16E2CABC2E6CBEBC0035390E.sh (in target 'ReduxDemo' from project 'ReduxDemo')
+    cd /Users/john/repositories/SwiftUI-Redux-Demo
+    /bin/sh -c /Users/john/Library/Developer/Xcode/DerivedData/ReduxDemo-gttjropgetipjmadcueprkkeczpt/Build/Intermediates.noindex/ReduxDemo.build/Debug-iphonesimulator/ReduxDemo.build/Script-16E2CABC2E6CBEBC0035390E.sh
+Phase
+Command PhaseScriptExecution failed with a nonzero exit code
+
+
+Result bundle written to path:
+	/Users/john/repositories/SwiftUI-Redux-Demo/.nvim/xcodebuild/build.xcresult
+
+** BUILD FAILED **
+
+
+The following build commands failed:
+	PhaseScriptExecution MY\ CUSTOM\ PHASE /Users/john/Library/Developer/Xcode/DerivedData/ReduxDemo-gttjropgetipjmadcueprkkeczpt/Build/Intermediates.noindex/ReduxDemo.build/Debug-iphonesimulator/ReduxDemo.build/Script-16E2CABC2E6CBEBC0035390E.sh (in target 'ReduxDemo' from project 'ReduxDemo')
+	Building project ReduxDemo with scheme ReduxDemo
+(2 failures)
+

--- a/specs/parser_test_data/tc24_out.log
+++ b/specs/parser_test_data/tc24_out.log
@@ -1,0 +1,13 @@
+{
+  buildErrors = { {
+      message = { "Build phase 'MY CUSTOM PHASE' failed:", "Command PhaseScriptExecution failed with a nonzero exit code" }
+    } },
+  buildWarnings = {},
+  failedTestsCount = 0,
+  output = {},
+  testErrors = {},
+  tests = {},
+  testsCount = 0,
+  usesSwiftTesting = false,
+  xcresultFilepath = "/Users/john/repositories/SwiftUI-Redux-Demo/.nvim/xcodebuild/build.xcresult"
+}

--- a/specs/parser_test_data/tc25.log
+++ b/specs/parser_test_data/tc25.log
@@ -1,0 +1,45 @@
+
+Command line invocation:
+    /Applications/Xcode.app/Contents/Developer/usr/bin/xcodebuild build -project /Users/john/repositories/SwiftUI-Redux-Demo/ReduxDemo.xcodeproj -scheme ReduxDemo -destination id=B1982473-232A-4810-A4B0-E71E4516F39B -resultBundlePath /Users/john/repositories/SwiftUI-Redux-Demo/.nvim/xcodebuild/build.xcresult -parallelizeTargets
+
+Writing result bundle at path:
+	/Users/john/repositories/SwiftUI-Redux-Demo/.nvim/xcodebuild/build.xcresult
+
+--- xcodebuild: WARNING: Using the first of multiple matching destinations:
+{ platform:iOS Simulator, arch:arm64, id:B1982473-232A-4810-A4B0-E71E4516F39B, OS:18.5, name:iPhone 16 }
+{ platform:iOS Simulator, arch:x86_64, id:B1982473-232A-4810-A4B0-E71E4516F39B, OS:18.5, name:iPhone 16 }
+ComputePackagePrebuildTargetDependencyGraph
+
+Prepare packages
+
+CreateBuildRequest
+
+SendProjectDescription
+
+CreateBuildOperation
+
+ComputeTargetDependencyGraph
+note: Building targets in dependency order
+note: Target dependency graph (1 target)
+    Target 'ReduxDemo' in project 'ReduxDemo' (no dependencies)
+
+GatherProvisioningInputs
+
+CreateBuildDescription
+
+ClangStatCache /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang-stat-cache /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator18.5.sdk /Users/john/Library/Developer/Xcode/DerivedData/SDKStatCaches.noindex/iphonesimulator18.5-22F76-d5fc8ad4295d2ef488fb7d0f804ce0c4.sdkstatcache
+    cd /Users/john/repositories/SwiftUI-Redux-Demo/ReduxDemo.xcodeproj
+    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang-stat-cache /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator18.5.sdk -o /Users/john/Library/Developer/Xcode/DerivedData/SDKStatCaches.noindex/iphonesimulator18.5-22F76-d5fc8ad4295d2ef488fb7d0f804ce0c4.sdkstatcache
+
+warning: Run script build phase 'MY CUSTOM PHASE' will be run during every build because it does not specify any outputs. To address this issue, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'ReduxDemo' from project 'ReduxDemo')
+PhaseScriptExecution MY\ CUSTOM\ PHASE /Users/john/Library/Developer/Xcode/DerivedData/ReduxDemo-gttjropgetipjmadcueprkkeczpt/Build/Intermediates.noindex/ReduxDemo.build/Debug-iphonesimulator/ReduxDemo.build/Script-16E2CABC2E6CBEBC0035390E.sh (in target 'ReduxDemo' from project 'ReduxDemo')
+    cd /Users/john/repositories/SwiftUI-Redux-Demo
+    /bin/sh -c /Users/john/Library/Developer/Xcode/DerivedData/ReduxDemo-gttjropgetipjmadcueprkkeczpt/Build/Intermediates.noindex/ReduxDemo.build/Debug-iphonesimulator/ReduxDemo.build/Script-16E2CABC2E6CBEBC0035390E.sh
+Phase
+
+
+Result bundle written to path:
+	/Users/john/repositories/SwiftUI-Redux-Demo/.nvim/xcodebuild/build.xcresult
+
+** BUILD FAILED **
+

--- a/specs/parser_test_data/tc25_out.log
+++ b/specs/parser_test_data/tc25_out.log
@@ -1,0 +1,13 @@
+{
+  buildErrors = { {
+      message = { "** BUILD FAILED **" }
+    } },
+  buildWarnings = {},
+  failedTestsCount = 0,
+  output = {},
+  testErrors = {},
+  tests = {},
+  testsCount = 0,
+  usesSwiftTesting = false,
+  xcresultFilepath = "/Users/john/repositories/SwiftUI-Redux-Demo/.nvim/xcodebuild/build.xcresult"
+}

